pkg/addr/addr.go

```diff
@@ -57,62 +59,38 @@ func ProxyAddressesToString(addrs []pb.TcpAddress) string {
 
 // ProxyIPToString formats a Proxy API IPAddress as a string.
 func ProxyIPToString(ip *pb.IPAddress) string {
-	octets := decodeIPToOctets(ip.GetIpv4())
-	return fmt.Sprintf("%d.%d.%d.%d", octets[0], octets[1], octets[2], octets[3])
-}
-
-// ProxyIPV4 encodes 4 octets as a Proxy API IPAddress.
-func ProxyIPV4(a1, a2, a3, a4 uint8) *pb.IPAddress {
-	ip := (uint32(a1) << 24) | (uint32(a2) << 16) | (uint32(a3) << 8) | uint32(a4)
-	return &pb.IPAddress{
-		Ip: &pb.IPAddress_Ipv4{
-			Ipv4: ip,
-		},
-	}
+	netIP := decodeIPv4ToNetIP(ip.GetIpv4())
+	return netIP.String()
 }
 
 // ParseProxyIPV4 parses an IP Address string into a Proxy API IPAddress.
 func ParseProxyIPV4(ip string) (*pb.IPAddress, error) {
-	segments := strings.Split(ip, ".")
-	if len(segments) != 4 {
-		return nil, fmt.Errorf("Invalid IP address: %s", ip)
-	}
-	octets := [4]uint8{0, 0, 0, 0}
-	for i, segment := range segments {
-		octet, err := strconv.ParseUint(segment, 10, 8)
-		if err != nil {
-			return nil, fmt.Errorf("Invalid IP segment: %s", segment)
+	netIP := net.ParseIP(ip)
+	if netIP != nil {
+		oBigInt := IPToInt(netIP.To4())
+		netIPAddress := &pb.IPAddress{
+			Ip: &pb.IPAddress_Ipv4{
+				Ipv4: uint32(oBigInt.Uint64()),
+			},
 		}
-		octets[i] = uint8(octet)
-	}
-	return ProxyIPV4(octets[0], octets[1], octets[2], octets[3]), nil
-}
-
-// PublicIPV4 encodes 4 octets as a Viz API IPAddress.
-func PublicIPV4(a1, a2, a3, a4 uint8) *l5dNetPb.IPAddress {
-	ip := (uint32(a1) << 24) | (uint32(a2) << 16) | (uint32(a3) << 8) | uint32(a4)
-	return &l5dNetPb.IPAddress{
-		Ip: &l5dNetPb.IPAddress_Ipv4{
-			Ipv4: ip,
-		},
+		return netIPAddress, nil
 	}
+	return nil, fmt.Errorf("Invalid IP address: %s", ip)
```

Comment:
nit: we could also possibly re-write this as

```suggestion
   if netIP == nil {
     return nil, fmt.Errorf("Invalid IP address: %s", ip)
   }
   
   oBigInt := IPToInt(netIP.To4())
   return &pb.IPAddress{
      Ip: &pb.IPAddress_Ipv4{
         Ipv4: uint32(oBigInt.Uint64()))
      }
   }
```

although not necessary. It might help make the error a bit more obvious, but I think the function is obvious as it is now too, will leave it up to you :)

Reply:
Hi @mateiidavid, that's a good idea, I change it as you suggest, thanks!